File: controller/usercontroller.js
const User = require('../models/user')
const bcrypt = require('bcryptjs')
const jwt=  require('jsonwebtoken')
const saltRounds = 5

exports.register = async (req,res)=>{
    try{
        const {email,name, password} = req.body;
        
        const user = await User.findOne({name : name})
        if(!user){
            var today = new Date();
            var dd = String(today.getDate()).padStart(2, '0');
            var mm = String(today.getMonth() + 1).padStart(2, '0'); //January is 0!
            var yyyy = today.getFullYear();

            today = mm + '/' + dd + '/' + yyyy;
            let hashed_password = await bcrypt.hash(password, saltRounds);
            // bcrypt.hash(password, saltRounds, (err, hash) {
                let new_user = new User({
                    name : name,
                    email : email,
                    password : hashed_password,
                    history : "",
                    createdAt : today
                })
            // });       
            await new_user.save();
            res.status(200).send({user : new_user })
        }else{
            throw("already exists")
        }
    }catch(err){
        res.status(400).send({message : err})
    }
}


exports.login = async(req,res)=>{
    try{
        const {email,password} = req.body;

        console.log(req.body)

        //check for email if exists

        let user = await User.findOne({email : email});
        
        if(!user){
            throw("Does not exists")
        }
        // it exists
        const issame = await bcrypt.compare(password, user.password);
        if(!issame){
            throw("Password error")
        }
        const token = jwt.sign({user},'lodekasignature')
        res.status(200).cookie('token',token, { maxAge: 36000, httpOnly: true }).json({userdetial : {
           email : user.email ,
           name : user.name
        }})

    }catch(err){
        res.status(400).json({message : err})
    }
}

File: backend/controller/usercontroller.py
from flask import Blueprint, request, jsonify, make_response
from werkzeug.security import generate_password_hash, check_password_hash
import datetime
from models.user import User # Assuming you've defined a User model with SQLAlchemy or any ORM you're using
from utils.connectdb_utils import connectDb
import jwt
import app

user_blueprint = Blueprint('user_bp', __name__)
saltRounds = 5
mongodb = connectDb()
user_collection = mongodb['users']
@user_blueprint.route('/register', methods=['POST'])
def register():
    try:
        data = request.get_json()
        email = data.get('email')
        name = data.get('name')
        password = data.get('password')
        print(data)
        user = user_collection.find_one({'email':email})
        print("User  : ",user)

        if user:
            return jsonify({"message": "already exists"}), 400

        today = datetime.datetime.now().strftime('%m/%d/%Y')
        hashed_password = generate_password_hash(password)

        new_user = User(
            name=name,
            email=email,
            password=hashed_password,
            history=""
        )

        new_user.save()
        return jsonify({"user": new_user.to_dict()}), 200  # Assuming your model has a `to_dict()` method

    except Exception as err:
        print(err)
        return jsonify({"message": str(err)}), 400


@user_blueprint.route('/login', methods=['POST'])
def login():
    try:
        data = request.json
        email = data.get('email')
        password = data.get('password')

        user = user_collection.find_one({"email":email})

        if not user:
            return jsonify({"message": "Does not exist"}), 400

        if not check_password_hash(user['password'], password):
            return jsonify({"message": "Password error"}), 400
        token = jwt.encode({'user_id': str(user['_id']), 'exp': datetime.datetime.utcnow() + datetime.timedelta(hours=1)},key= app.app.secret_key)
        response = make_response(jsonify({"userdetail": {"email": user['email'], "name": user['name'] ,"token":token}}), 200)               
        return response

    except Exception as err:
        return jsonify({"message": str(err)}), 400


